Mlem/App/Views/Shared/TappableImageView.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  TappableImageView.swift
+//  Mlem
+//
+//  Created by Sjmarf on 13/06/2024.
+//
+
+import SwiftUI
+
+struct TappableImageView: View {
+    @Environment(NavigationLayer.self) var navigation
+    
+    @State var loading: ImageView.LoadingState = .waiting
+    let url: URL?
+    
+    var body: some View {
+        ImageView(url: url, onLoadingStateChange: { loading = $0 })
+            .onTapGesture {
+                if loading == .done, let url {
+                    // Sheets don't cover the whole screen on iPad, so use a fullScreenCover instead
+                    if UIDevice.isIPad {
+                        navigation.showFullScreenCover(.imageViewer(url))
+                    } else {
+                        navigation.openSheet(.imageViewer(url))
+                    }
```

Comment:
I doubt this will be the only occurrence of this pattern; we may wish to define a `multiplatformAction` function analogous to the `MultiplatformView`. Possibly OOS for this PR, content to merge as-is.

Reply:
Would that be particularly beneficial? It would be more verbose to do it that way I think?

```swift
multiPlatformAction(pad: {
    navigation.showFullScreenCover(.imageViewer(url))
}, phone: {
    navigation.openSheet(.imageViewer(url))
})
```

vs

```swift
if UIDevice.isPad {
    navigation.showFullScreenCover(.imageViewer(url))
} else {
    navigation.openSheet(.imageViewer(url))
}
```